libs/sdk-core/src/reverseswap.rs

```diff
@@ -22,68 +37,342 @@ pub struct CreateReverseSwapResponse {
+                        EcdsaSighashType::All,
+                    )?;
+                    let msg = Message::from_slice(&sig[..])?;
+                    let secret_key = SecretKey::from_slice(rs.local_private_key.as_slice())?;
+                    let sig = scpt.sign_ecdsa(&msg, &secret_key);
+
+                    let mut sigvec = sig.serialize_der().to_vec();
+                    sigvec.push(EcdsaSighashType::All as u8);
+
+                    let witness: Vec<Vec<u8>> = vec![
+                        sigvec,
+                        rs.local_preimage.clone(),
+                        claim_script_bytes.clone(),
+                    ];
+
+                    let mut signed_input = input.clone();
+                    let w = Witness::from_vec(witness);
+                    signed_input.witness = w;
+                    signed_inputs.push(signed_input);
+                }
+                tx.input = signed_inputs;
+
+                Ok(tx)
+            }
+            Some(addr_type) => Err(anyhow!("Unexpected lock address type: {addr_type:?}")),
+            None => Err(anyhow!("Could not determine lock address type")),
+        }
+    }
+
+    /// Executes the corresponding next steps in the pending reverse swaps.
+    ///
+    /// Expects recently refreshed rev swap statuses to be present in the DB cache
+    /// via [Self::refresh_monitored_reverse_swaps]
+    pub(crate) async fn execute_pending_reverse_swaps(&self) -> Result<()> {
+        let monitored = self.list_monitored().await?;
+        info!("Found {} monitored reverse swaps", monitored.len());
+
+        // Depending on the new state, decide next steps and transition to the new state
+        for rs in monitored {
+            info!("Checking monitored {rs:?}");
+
+            if matches!(
+                rs.cache.boltz_api_status,
+                BoltzApiReverseSwapStatus::LockTxConfirmed { .. }
+            ) {
+                info!("Lock tx is confirmed, preparing claim tx");
+                let claim_tx = self.create_claim_tx(&rs).await?;
+                let claim_tx_broadcast_res = self
+                    .chain_service
+                    .broadcast_transaction(bitcoin::psbt::serialize::Serialize::serialize(
+                        &claim_tx,
+                    ))
+                    .await;
+                info!("Broadcast claim tx result: {claim_tx_broadcast_res:?}");
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Updates the state of monitored reverse swaps in the cache table. This includes the blocking
+    /// reverse swaps as well, since the blocking statuses are a subset of the monitored statuses.
+    async fn refresh_monitored_reverse_swaps(&self) -> Result<()> {
+        let to_check = self.list_monitored().await?;
+        for rs in to_check {
+            let id = rs.id.clone();
+            let api = self.reverse_swapper_api.clone();
+            let fresh_boltz_status = api.get_dynamic_boltz_status(id.clone()).await?;
```

Comment:
Why do we need to get the status from Boltz? In theory, we can deduce the status from the LN payments and the onchain information. Perhaps we can keep the status only for logging.

Reply:
Done: 6303df600cb561dc8b0948cd2dae96c606271992